### PR TITLE
Triton Kernels for rest of Transformer ops

### PR DESCRIPTION
**Pull request by martin-kukla**
- Added Triton Kernels for rest of Transformer's ops (matmul, FlashAttentionv1)
- Improved some of the existing Triton Kernels (e.g. Cross Entropy loss)
- There are some numerical differences between this and `torch_func+autograd+jit`. They are rather small, but to be investigated
- The current speed comparison:
  -  `triton`: **3.15it/s** 
  -  `torch_func+autograd+jit`: **4.06it/s**
- The current memory comparison:
  -  `triton`: **7GB**
  -  `torch_func+autograd+jit`: **21GB**

### CODE REVIEW

**Comment by martin-kukla** [comment]
Here is the training loss comparison of `triton` and `torch_func+autograd+jit`

Screenshot 2025-02-20 at 04 39 20
